### PR TITLE
Some misc tweaks regarding eg. tap

### PR DESCRIPTION
**Pull request by voxpelli**
I had trouble getting `tap` running locally on my node `16.11.1` + npm `8.1.1`, not even the `--legacy-bundling=true` trick helped.

So, I did two things:

* Updated to lkatest version of `tap`
* Swapped the reporter to one which doesn't require `react`(!?)

And as an added fix:

* I made it so that no `package-lock.json` gets generated locally

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,8 @@
   "scripts": {
     "lint": "standard | snazzy",
     "lint-ci": "standard",
-    "test": "tap --no-cov 'test/**/*.test.js'",
-    "test-ci": "tap --cov --coverage-report=text 'test/**/*.test.js'",
+    "test": "tap --no-cov -Rtap 'test/**/*.test.js'",
+    "test-ci": "tap --cov --coverage-report=text -Rtap 'test/**/*.test.js'",
```
Comment (jsumners) [inline package.json:12]: If we add `legacy-peer-deps=true` to the `.npmrc`, then this shouldn't be necessary (given the update to `tap@15`.

Comment (voxpelli) [reply package.json]: @jsumners 🤯 yikes, now I really understand what you were talking about on Twitter. How can this fix it!? 😱 (It did though, pushed a new fix)

**Review by mcollina — APPROVED** [review]
lgtm

Just bumping the dependency and cleaning up node_modules fixes it

**Comment by voxpelli** [comment]
Looks like the coverage numbers suddenly doesn't match `100%` any more 🤔

**Comment by voxpelli** [comment]
Got confused by the code coverage flags and changes, the relevant change in `tap@15` is that it now has `--check-coverage` on by default, which `tap@14` didn't have.

Fixed by adding `--no-check-coverage`

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Pull request merged**